src/wallet/wallet.h

```diff
@@ -780,6 +780,7 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
         nRelockTime = 0;
         fAbortRescan = false;
         fScanningWallet = false;
+        fBalancesDirty = true;
```

Comment:
Oh. Yes indeed. I now switched to a cache-invalidation-atomic per balance type and set it to `false` once the cache has been built.